sympy/sets/tests/test_fancysets.py

```diff
@@ -295,18 +295,162 @@ def test_Range_set():
         assert r.inf == rev.inf and r.sup == rev.sup
         assert r.step == -rev.step
 
-    # Make sure to use range in Python 3 and xrange in Python 2 (regardless of
-    # compatibility imports above)
+    # test symbolic Range
+    a, b, c, d = symbols('a, b, c, d', integer=True)
+    r = Range(a, b, c)
+    rrev = r.reversed
+    avs, bvs = ([0, 1, -1, 2, -2], )*2
+    cvs = [1, -1, 2, -2]
+    check = lambda rabc, rabcr: (rabcr.inf == rabc.inf and
+                                 rabcr.sup == rabc.sup and
+                                 rabcr.step == -rabc.step)
```

Comment:
I'm sure that the tests from here down are very comprehensive but it's hard for me to tell from the diff here what they do. Maybe at least some comments would help here